lib/modules/platform/gitea/gitea-helper.ts

```diff
@@ -183,6 +183,21 @@ export async function getPR(
   return res.body;
 }
 
+export async function getPRByBranch(
+  repoPath: string,
+  base: string,
+  head: string,
+  options?: GiteaHttpOptions,
+): Promise<PR | null> {
+  const url = `${API_PATH}/repos/${repoPath}/pulls/${base}/${head}`;
+  try {
+    const res = await giteaHttp.getJsonUnchecked<PR>(url, options);
+    return res.body;
+  } catch {
+    return null;
```

Comment:
log error as trace for debug purpose. only ignore 404. if any other errors occur then log debug.

we should also use zod schema for new functions

Reply:
1st one done..

For the zod schema, we will need some discussion before about how to handle the `LongCommitSha` branded type. Because, we will first need to create a branded shcema in zod and infer type from it and then use that type everywhere instead if the TS type we use currently, for zod validation to be possible.

Reply:
Should I create an issue for that?

Reply:
yes please

Reply:
Created a discussion: https://github.com/renovatebot/renovate/discussions/35099